go/ethclient/interface.go

```diff
@@ -23,9 +23,10 @@ type EthClient interface {
 	IsBlockAncestor(block *types.Block, proof obscurocommon.L1RootHash) bool
 
 	RPCBlockchainFeed() []*types.Block                             // returns all blocks from genesis to head
-	BroadcastTx(t *obscurocommon.L1TxData)                         // issues an obscurocommon.L1TxData to the L1 network
+	BroadcastTx(tx types.TxData)                                   // issues an obscurocommon.L1Transaction to the L1 network
 	BlockListener() chan *types.Header                             // subscribes to new blocks and returns a listener with the blocks heads
-	SubmitTransaction(tx types.TxData) (*types.Transaction, error) // submits an ethereum transaction
+	SubmitTransaction(tx types.TxData) (*types.Transaction, error) // signs and issues an ethereum transaction
```

Comment:
it's quite unclear how BroadcastTx, SubmitTransaction and IssueTransaction differ from each other.
I'd suggest a clean-up of the interface.
I don't think "SubmitTransaction" and "IssueTransaction" need to be on this interface at all.

Reply:
Refactored those 👍